python/lsst/rubintv/production/uploaders.py

```diff
@@ -20,37 +20,658 @@
+
+        Returns
+        -------
+        s3bucket : `ServiceResource`
+            S3 bucket ready to use for file transfer
+        """
+        try:
+            proxyDict = {"http": proxyUrl, "https": proxyUrl}
+
+            # Create a custom botocore session
+            session = S3_session(profile_name=bucketInfo.profileName)
+
+            # Create an S3 resource with the custom botocore session
+            s3Resource = session.resource('s3', endpoint_url=endPoint,
+                                          config=Config(proxies=proxyDict))
+            return s3Resource.Bucket(bucketInfo.bucketName)
+        except ClientError:
+            raise ConnectionError(f"Failed client connection: {bucketInfo.profileName}")
+
+    def uploadPerSeqNumPlot(
+        self,
+        instrument: str,
+        plotName: str,
+        dayObs: int,
+        seqNum: int,
+        filename: str,
+    ) -> str:
+        """Upload a per-dayObs/seqNum plot to the bucket.
+
+        Parameters
+        ----------
+        instrument : `str`
+            The instrument the plot is for.
+        plotName : `str`
+            The name of the plot.
+        dayObs : `int`
+            The dayObs of the plot.
+        seqNum : `int`
+            The seqNum of the plot.
+        filename : `str`
+            The full path and filename of the file to upload.
+
+        Raises
+        ------
+        ValueError
+            Raised if the specified channel is not in the list of existing
+            channels as specified in CHANNELS.
+        UploadError
+            Raised if uploading the file to the Bucket was not possible.
+
+        Returns
+        -------
+        uploadAs: `str``
+            Path and filename for the destination file in the bucket
+        """
+        checkInstrument(instrument)
+
+        dayObsStr = dayObsIntToString(dayObs)
+        paddedSeqNum = f"{seqNum:06}"
+        extension = os.path.splitext(filename)[1]  # contains the period so don't add back later
+        uploadAs = (
+            f"{instrument}"
+            f"/{dayObsStr}"
+            f"/{plotName}"
+            f"/{paddedSeqNum}"
+            f"/{instrument}_{plotName}_{dayObsStr}_{paddedSeqNum}{extension}"
+        )
+
+        try:
+            self.upload(destinationFilename=uploadAs, sourceFilename=filename)
+            self._log.info(f"Uploaded {filename} to {uploadAs}")
+        except Exception as e:
+            self._log.exception(
+                f"Failed to upload {filename} for {instrument}+{plotName} as {uploadAs}"
+            )
+            raise e
+
+        return uploadAs
+
+    @override
+    def uploadNightReportData(
+        self,
+        instrument: str,
+        dayObs: int,
+        filename: str,
+        plotGroup: str = None,
+    ) -> str:
+        """Upload night report type plot or json file
+           to a night report channel.
+
+        Parameters
+        ----------
+        instrument : `str`
```

Comment:
Note throughout: since you're using type hints, there's no need to put the type of each parameter in the docstring. But there are a couple of classes/functions that you aren't using type hints, but maybe you should...